src/settings/settingsdialog.cpp

```diff
@@ -379,22 +388,53 @@ void SettingsDialog::saveYandexEngineEmotion(int emotion)
 // To play test text
 void SettingsDialog::detectYandexTextLanguage()
 {
-    if (ui->yandexTestSpeechEdit->text().isEmpty()) {
-        QMessageBox::information(this, tr("Nothing to play"), tr("Playback text is empty"));
-        return;
-    }
-
-    m_yandexTranslator->detectLanguage(ui->yandexTestSpeechEdit->text(), QOnlineTranslator::Yandex);
+    detectTestTextLanguage(m_yandexTranslator, QOnlineTranslator::Yandex);
 }
 
 void SettingsDialog::speakYandexTestText()
 {
-    if (m_yandexTranslator->error() != QOnlineTranslator::NoError) {
-        QMessageBox::critical(this, tr("Unable to detect language"), m_yandexTranslator->errorString());
-        return;
+    speakTestText(m_yandexTranslator, QOnlineTranslator::Yandex);
+}
+
+void SettingsDialog::onGoogleLanguageSelectionChanged(int languageIndex)
+{
+    auto const configuredLang = ui->googleLanguageComboBox->itemData(languageIndex).value<QOnlineTranslator::Language>();
+    auto const originalRegion = ui->googlePlayerButtons->regions(QOnlineTranslator::Google)[configuredLang]; // It will be lost after googleRegionComboBox is changed if not stored here
```

Comment:
Please, use exact types here and in other places. We use `auto` only if variable type is obvious from the context. Read [this](https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-auto.html) for details.

Reply:
Will do. My style definitely needs polishing.